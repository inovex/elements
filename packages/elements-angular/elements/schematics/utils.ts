import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
