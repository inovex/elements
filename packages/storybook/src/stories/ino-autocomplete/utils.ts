import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
