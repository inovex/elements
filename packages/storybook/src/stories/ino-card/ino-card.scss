.story-card {
  .customizable-card {
    margin-top: 12px;
  }

  .flex-cards {
    margin-top: 50px;
    display: flex;
    justify-content: space-between;
    align-items: center;

    > ino-card {
      flex: 1;
      margin-right: 10px;
      &:last-child {
        margin-right: 0;
      }
    }
  }

  h4 {
    margin-top: 52px;
  }
}

.card-content--vertical {
  display: flex;
  flex-direction: row;
  justify-content: center;

  h3 {
    margin: 0;
  }

  ino-img img {
    width: 100%;
    height: 100%;
  }
}

.card-header {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
}

.sample-card-footer {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
}

.card-footer {
  display: flex;
  flex-direction: row;
  justify-content: center;

  p {
    color: #c1c1c1;
    margin: 0;
  }
}

ino-card {
  width: 300px;
}
