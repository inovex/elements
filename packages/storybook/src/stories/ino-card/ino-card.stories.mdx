import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-card/readme.md";

<Meta title="Structure/ino-card/Documentation" />

<Description markdown={readme} />
