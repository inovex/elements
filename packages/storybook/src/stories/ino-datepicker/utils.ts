import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
