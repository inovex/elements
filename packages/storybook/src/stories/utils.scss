/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
