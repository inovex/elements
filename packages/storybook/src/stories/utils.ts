import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
