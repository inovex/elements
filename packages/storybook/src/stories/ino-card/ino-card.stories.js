import { boolean, select } from '@storybook/addon-knobs';

import componentReadme from '_local-elements/src/components/ino-card/readme.md';
import withStencilReadme from '_local-storybookcore/with-stencil-readme';

import './ino-card.scss';

function subscribeToComponentEvents() {
  // == event block
  const eventHandler = function(e) {
    const el = e.target;
    if (el === null) {
      return;
    }
    const card = el.closest('ino-card');
    if (card === null) {
      return;
    }
    card.inoSelected = !card.inoSelected;
  };

  document.addEventListener('click', eventHandler);
  // == event block

  // unsubscribe function will be called by Storybook
  return () => {
    document.removeEventListener('click', eventHandler);
  };
}

export default {
  title: 'Structure/<ino-card>',

  decorators: [
    withStencilReadme(componentReadme),
    story => {
      subscribeToComponentEvents();
      return story();
    }
  ]
};

export const DefaultUsage = () => /*html*/ `
    <div class="story-card">

      <h4>Customizable card</h4>
      <ino-card class="customizable-card"
        ino-disable-elevation="${boolean('ino-disable-elevation', false)}"
        ino-selected="${boolean('ino-selected', false)}">
        ${sampleCard}
      </ino-card>

      <div class="flex-cards">
        ${flexCard}
        ${flexCard}
        ${flexCard}
      </div>
    </div>
  `;

const sampleCard = `
  <div slot="header" class="card-header">
      <ino-icon ino-icon="info" ino-clickable></ino-icon>
  </div>
  <div slot="content" class="card-content--vertical">
      <ino-img src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png"></ino-img>
  </div>
  <div slot="footer" class="sample-card-footer">
    <ino-segment-group>
        <ino-segment-button value="1">Read</ino-segment-button>
        <ino-segment-button value="2">Bookmark</ino-segment-button>
    </ino-segment-group>
  </div>
`;

const flexCard = `
  <ino-card>
      <div slot="header" class="card-header">
          <ino-icon ino-icon="info" ino-clickable></ino-icon>
      </div>
      <div slot="content" class="card-content--vertical">
          <h3>Primary</h3>
      </div>
      <div slot="footer" class="card-footer">
          <p>Subtitle</p>
      </div>
  </ino-card>
`;
