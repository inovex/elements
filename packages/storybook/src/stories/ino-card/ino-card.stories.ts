import { Components } from '@inovex.de/elements';
import { useEffect } from '@storybook/client-api';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';
import { defaultDecorator } from '../utils';

import './ino-card.scss';

export default {
  title: 'Structure/<ino-card>',
  component: 'ino-card',
  decorators: [
    defaultDecorator,
    (story) => {
      useEffect(() => {
        const handleClick = function (e) {
          const maybeCard: HTMLInoCardElement | undefined = e.target?.closest(
            'ino-card'
          );

          if (!maybeCard === null) return;

          maybeCard.selected = !maybeCard.selected;
        };

        document.addEventListener('click', handleClick);

        return () => {
          document.removeEventListener('click', handleClick);
        };
      });

      return story();
    },
  ],
};

export const Playground: Story<Components.InoCard> = (args) => html`
  <ino-card
    class="customizable-card"
    disable-elevation="${args.disableElevation}"
    selected="${args.selected}"
  >
    <div slot="header" class="card-header">
      <ino-icon icon="info" clickable></ino-icon>
    </div>
    <div slot="content" class="card-content--vertical">
      <ino-img
        src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png"
      ></ino-img>
    </div>
    <div slot="footer" class="sample-card-footer">
      <ino-segment-group>
        <ino-segment-button value="1">Read</ino-segment-button>
        <ino-segment-button value="2">Bookmark</ino-segment-button>
      </ino-segment-group>
    </div>
  </ino-card>
`;

Playground.args = {
  disableElevation: false,
  selected: false,
};
