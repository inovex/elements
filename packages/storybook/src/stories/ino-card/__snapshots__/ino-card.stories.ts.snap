// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/<ino-card> DisabledElevation smoke-test 1`] = `
<div id="root-inner">
  <div class="story-card">
    <ino-card class="customizable-card hydrated"
              disable-elevation="true"
              selected="false"
    >
      <div class="mdc-card mdc-card--outlined">
        <div class="ino-card__checkmark-container">
        </div>
        <div class="ino-card__header">
          <div slot="header"
               class="card-header"
          >
            <ino-icon icon="info"
                      clickable
                      class="ino-icon--color-scheme-primary hydrated"
            >
            </ino-icon>
          </div>
        </div>
        <div class="ino-card__content">
          <div slot="content"
               class="card-content--vertical"
          >
            <ino-img src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png"
                     class="hydrated"
            >
              <div class="ino-img__composer"
                   style="padding-top: 100%;"
              >
                <img class="ino-img__image"
                     src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png"
                >
                <div class="ino-img__fallback-icon">
                  <ino-icon class="ino-icon--color-scheme-primary hydrated">
                  </ino-icon>
                </div>
              </div>
            </ino-img>
          </div>
        </div>
        <div class="ino-card__footer">
          <div slot="footer"
               class="sample-card-footer"
          >
            <ino-segment-group id="ino-segment-group-id__cg0cfpq5y"
                               class="hydrated"
            >
              <ino-segment-button value="1"
                                  class="hydrated"
              >
                Read
              </ino-segment-button>
              <ino-segment-button value="2"
                                  class="hydrated"
              >
                Bookmark
              </ino-segment-button>
            </ino-segment-group>
          </div>
        </div>
      </div>
    </ino-card>
  </div>
</div>
`;

exports[`Structure/<ino-card> Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-card">
    <ino-card class="customizable-card ino-elevate hydrated"
              disable-elevation="false"
              selected="false"
    >
      <div class="mdc-card mdc-card--outlined">
        <div class="ino-card__checkmark-container">
        </div>
        <div class="ino-card__header">
          <div slot="header"
               class="card-header"
          >
            <ino-icon icon="info"
                      clickable
                      class="hydrated ino-icon--color-scheme-primary"
            >
            </ino-icon>
          </div>
        </div>
        <div class="ino-card__content">
          <div slot="content"
               class="card-content--vertical"
          >
            <ino-img src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png"
                     class="hydrated"
            >
              <div class="ino-img__composer"
                   style="padding-top: 100%;"
              >
                <img class="ino-img__image"
                     src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png"
                >
                <div class="ino-img__fallback-icon">
                  <ino-icon class="hydrated ino-icon--color-scheme-primary">
                  </ino-icon>
                </div>
              </div>
            </ino-img>
          </div>
        </div>
        <div class="ino-card__footer">
          <div slot="footer"
               class="sample-card-footer"
          >
            <ino-segment-group id="ino-segment-group-id__za0fcf6ur"
                               class="hydrated"
            >
              <ino-segment-button value="1"
                                  class="hydrated"
              >
                Read
              </ino-segment-button>
              <ino-segment-button value="2"
                                  class="hydrated"
              >
                Bookmark
              </ino-segment-button>
            </ino-segment-group>
          </div>
        </div>
      </div>
    </ino-card>
  </div>
</div>
`;
